pytorch_lightning/strategies/deepspeed.py

```diff
@@ -357,6 +357,8 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
         self.setup_optimizers(trainer)
         self.setup_precision_plugin()
+        if self.root_device == torch.device("cpu"):
```

Comment:
should we raise this error on init? cause when someone initiates the trainer, we don't change the accelerator/strategy after that, so we can simply check this during init.